app/_plugins/generators/plugin_single_source_generator.rb

```diff
@@ -0,0 +1,90 @@
+module PluginSingleSource
+  class Generator < Jekyll::Generator
+    priority :highest
+    def generate(site)
+      site.data['ssg_hub'] = []
+      seen = []
+
+      Dir.glob('app/_data/extensions/*/*/versions.yml').each do|f|
+        name = f.gsub("app/_data/extensions/", "").gsub("/versions.yml","")
+        seen << name
+        data = SafeYAML.load(File.read(f))
+        createPages(data, site, name)
+      end
+
+      # Add any files that did not have a corresponding versions.yml
+      Dir.glob('app/_hub/*/*/_index.md').each do|f|
+        name = f.gsub("app/_hub/", "").gsub("/_index.md","")
+        next if seen.include?(name)
+        createPages([{'release' => "1.0.0"}], site, name) # If there's no version, assume it's 1.0.0
```

Comment:
I get this warning when I build the site locally:

```
DidYouMean::SPELL_CHECKERS.merge!(error_name => spell_checker)' has been deprecated. Please call `DidYouMean.correct_error(error_name, spell_checker)' instead
```
I _think_ it's referring to this line, though it doesn't specify.

Reply:
This is a side effect of another change that's on `main` (Ruby version upgrade).

It's local to your machine, and can be fixed with:

```
gem install bundler
bundle update --bundler
```